### PR TITLE
feat(Admin) : add Verify Tutor View

### PR DESCRIPTION
**Pull request by beesk135**

### CODE REVIEW

File: src/views/maintenance/Verify.vue [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        .then(res => {
+          if (res.data.verified) {
+            item.verified = true;
+            this.pushNotification("success", "Verify Succeeded");
+            this.save(item);
+          }
+        })
+        .catch(err => {
+          this.pushNotification("error", "Verify Not Suceeded");
+        });
```
Comment (UtopiaBeam) [inline src/views/maintenance/Verify.vue]: Use `async/await`

File: src/types/snackbar.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export enum SnackbarActions {
+  push = "pushNewNotification",
+  reset = "resetOpenState"
+}
+
+export enum SnackbarMutations {
+  set = "setSnackbarState",
+  reset = "resetSnackbarState"
+}
```
Comment (UtopiaBeam) [inline src/types/snackbar.ts]: Please change the key to match its value.

File: src/components/SnackbarNotification.vue [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+@Component
+export default class SnackbarNotification extends Vue {
+  @State(state => state.snackbar.isOpen) isOpen!: boolean;
+  @State(state => state.snackbar.color) color!: string;
+  @State(state => state.snackbar.message) message!: string;
+}
+</script>
+
+<style scoped></style>
```
Comment (UtopiaBeam) [inline src/components/SnackbarNotification.vue]: This can be removed

File: src/components/ReportSystemButton.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  dismissAndClearInput() {
+    this.title = "";
+    this.description = "";
+    this.dialog = false;
+  }
+}
+</script>
+
+<style scoped></style>
```
Comment (UtopiaBeam) [inline src/components/ReportSystemButton.vue]: This can be removed.

File: src/types/snackbar.ts (hunk 2 above)
Comment (beesk135) [reply src/types/snackbar.ts]: not my work

File: src/components/SnackbarNotification.vue (hunk 3 above)
Comment (beesk135) [reply src/components/SnackbarNotification.vue]: not my work

File: src/components/ReportSystemButton.vue (hunk 4 above)
Comment (beesk135) [reply src/components/ReportSystemButton.vue]: not my scope

File: src/views/maintenance/Verify.vue (hunk 1 above)
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+      if (this.editedIndex > -1) {
+        Object.assign(this.teachers[this.editedIndex], item);
+      }
+    },
+    pushNotification(color, messages) {
+      this.pushNewNotification({ color: color, message: messages });
+      this.dismissAndClearInput();
+    },
+    download(item) {
+      alert("got evidence file of ",);
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+            save(item);
+          }
+        })
+        .catch(err => {
+          this.pushNotification("error", "Verify Not Suceeded");
+        });
+    },
+    save(item) {
+      this.editedIndex = this.teachers.indexOf(item);
+      //this.editedItem = Object.assign({}, item)
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+    transform: rotate(360deg);
+  }
+}
+.mytable .v-table tbody tr {
+  background-color: black;
+  border-bottom: none !important;
+}
+// .mytable .v-table tbody tr:not(:last-child) {
+//     border-bottom: none;
+// }
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+    },
+    download(item) {
+      alert("got evidence file of ",);
+    }
+  },
+  mounted() {
+    Vue.axios.get("http://localhost:3000/admin/allTutor", config).then(res => {
+      this.teachers = res.data;
+      console.log(this.teachers);
+    });
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+        .then(res => {
+          if (!res.data.verified) {
+            item.verified = false;
+            this.pushNotification("success", "Unverify Succeeded");
+            save(item);
+          }
+        })
+        .catch(err => {
+          this.pushNotification("error", "Verify Not Suceeded");
+        });
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+  created() {
+    this.initialize();
+  },
+
+  methods: {
+    verifyTeacher(item) {
+      console.log(item);
+      Vue.axios
+        .post(
+          `http://localhost:3000/admin/verifyTutor/${item._id}`,
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/views/maintenance/Verify.vue [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+
+<script>
+import Vue from "vue";
+const token =
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZTY2ZDk0YWFhM2ZjODAwM2U4NjljMTYiLCJpYXQiOjE1ODM4NjUzNTB9.OYPPKY5x0WHe880t0JUiKNsCcBa-4rknPjHU_4MDkqo";
+const config = {
+  headers: {
+    Authorization: `Bearer ${token}`
+  }
+};
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/views/maintenance/Verify.vue]: ok

File: src/components/ReportSystemButton.vue [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+        this.dismissAndClearInput();
+      })
+      .catch(err => {
+        this.pushNewNotification({
+          color: "error",
+          message: "Report Submission Failed"
+        });
+        this.dismissAndClearInput();
+      });
+  }
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/components/ReportSystemButton.vue]: ok

**Comment by beesk135** [comment]
> `Verify` component should use `vue-property-decorator`. Please migrate.

wait me finish

File: src/store/modules/chat.ts [hunk 13]
```diff
@@ -76,7 +76,9 @@ const store: StoreOptions<ChatState> = {
     },
     [ChatAction.unsubscribe]: ({ commit }) => {
       commit(ChatMutation.setMessages, []);
-      const unsub = messageCol.onSnapshot(() => {});
+      const unsub = messageCol.onSnapshot(() => {
+        return {};
+      });
```
[the remark this answers is not in the document]
Comment (beesk135) [reply src/store/modules/chat.ts]: image

[the remark this answers is not in the document]
Comment (beesk135) [reply src/store/modules/chat.ts]: it showed sth like this which i could not ignore so i tried to remove that error

[the remark this answers is not in the document]
Comment (UtopiaBeam) [reply src/store/modules/chat.ts]: Nevermind, I will fix that.

File: src/components/Sidebar.vue [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    return this.menus.filter(menu => menu.public || this.isLogin);
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+#drawer {
+  margin-top: 60px;
+}
+</style>
```
Comment (theoatboy139) [inline src/components/Sidebar.vue]: remove `Sidebar.vue`. We no longer use this. It's already inside with `Navbar.vue`

File: src/views/Profile.vue [hunk 15]
```diff
@@ -86,17 +86,16 @@
...
   LoginActions,
   SignUpCredentials,
   UserGender,
-  UserRole,
   LoginGetters
 } from "../types";
-import { loginRules as rules, Rule } from "../rules";
-import vuetify from "../plugins/vuetify";
+import { loginRules as rules } from "../rules";
+// import vuetify from "../plugins/vuetify";
```
Comment (theoatboy139) [inline src/views/Profile.vue]: remove comment

Comment (beesk135) [reply src/views/Profile.vue]: Roger

File: src/views/Chats.vue [hunk 16]
```diff
@@ -20,14 +20,7 @@
 <script lang="ts">
 import { Vue, Component, Model, Watch } from "vue-property-decorator";
 import { Action, Getter } from "vuex-class";
-import {
-  ChatAction,
-  ChatUsers,
-  ChatGetters,
-  Message,
```
Comment (chavincc) [inline src/views/Chats.vue:27]: missing import

File: src/store/index.ts [hunk 17]
```diff
@@ -21,9 +22,7 @@ const store = new Vuex.Store<any>({
     chat,
     login,
     snackbar,
-    suspendUser,
-    appointment,
-    users
+    suspendUser
```
Comment (chavincc) [inline src/store/index.ts:25]: missing module

**Pull request merged**